web-api/src/Kerbero.Domain/SmartLockKeys/Interactors/CreateSmartLockKeyInteractor.cs

```diff
@@ -1,32 +1,54 @@
 using FluentResults;
-using Kerbero.Domain.SmartLockKeys.Managers;
+using Kerbero.Domain.NukiCredentials.Repositories;
+using Kerbero.Domain.SmartLockKeys.Dtos;
+using Kerbero.Domain.SmartLockKeys.Interfaces;
 using Kerbero.Domain.SmartLockKeys.Mappers;
-using Kerbero.Domain.SmartLockKeys.Models.PresentationRequests;
-using Kerbero.Domain.SmartLockKeys.Models.PresentationResponses;
 using Kerbero.Domain.SmartLockKeys.Repositories;
+using Kerbero.Domain.SmartLockKeys.Utils;
+using Kerbero.Domain.SmartLocks.Repositories;
 
 namespace Kerbero.Domain.SmartLockKeys.Interactors;
 
-public class CreateSmartLockKeyInteractor
+public class CreateSmartLockKeyInteractor: ICreateSmartLockKeyInteractor
 {
-	private readonly ISmartLockKeyGeneratorManager _smartLockKeyGeneratorManager;
-	private readonly ISmartLockKeyPersistentRepository _smartLockKeyPersistentRepository;
+	private readonly ISmartLockKeyRepository _smartLockKeyRepository;
+	private readonly INukiSmartLockRepository _nukiSmartLockRepository;
+	private readonly INukiCredentialRepository _nukiCredentialRepository;
 
-	public CreateSmartLockKeyInteractor(ISmartLockKeyGeneratorManager smartLockKeyGeneratorManager, ISmartLockKeyPersistentRepository smartLockKeyPersistentRepository)
+	public CreateSmartLockKeyInteractor(
+		ISmartLockKeyRepository smartLockKeyRepository,
+		INukiSmartLockRepository nukiSmartLockRepository,
+		INukiCredentialRepository nukiCredentialRepository)
 	{
-		_smartLockKeyGeneratorManager = smartLockKeyGeneratorManager;
-		_smartLockKeyPersistentRepository = smartLockKeyPersistentRepository;
+		_smartLockKeyRepository = smartLockKeyRepository;
+		_nukiSmartLockRepository = nukiSmartLockRepository;
+		_nukiCredentialRepository = nukiCredentialRepository;
 	}
 
-	public async Task<Result<CreateSmartLockKeyPresentationResponse>> Handle(CreateSmartLockKeyPresentationRequest presentationRequest)
+	public async Task<Result<SmartLockKeyDto>> Handle(CreateSmartLockKeyParams request)
 	{
-		var generatedKeyResult = _smartLockKeyGeneratorManager.GenerateSmartLockKey(presentationRequest.SmartLockId, presentationRequest.ExpiryDate);
+		// TODO switch on provider or move logic on another interactor
```

Comment:
```suggestion

```
can we check if the provider is not "nuki" and throw "UnsupportedSmartLockProvider" ?